doc/decisions/error_message.md

```diff
@@ -0,0 +1,215 @@
+    made up of 5 chars, where the first two chars indicate a certain class. This follows the SQLSTATE conventions. 
+    Currently they have 43 classes which all come from SQLSTATE. Postgres also throws additional errors but have to subclass it to one of the current 43 classes and have a special naming convention which have to start with a `P` in the subclass.
+* [etcd](https://github.com/etcd-io/etcd): 
+    Etcd's approach for errors are tightly coupled to the programming language Go as well as the [gRPC](https://grpc.io/) standard which currently has 
+    [16 codes](https://godoc.org/google.golang.org/grpc/codes) defined. Some of these errors are similar or identical to those which will be used in elektra.
+    Every error of etcd is associated with one of these categories and gets its own error message which is specified in [this](https://github.com/etcd-io/etcd/blob/master/etcdserver/api/v3rpc/rpctypes/error.go) file. This concept though does not allow easy subclassing which might be useful (eg. further split FailedPrecondition into more specific errors like semantic and syntactic errors)
+* [Windows Registry](https://docs.microsoft.com/en-us/windows/desktop/sysinfo/registry):
+    The registry does not use any specific error concept but takes the standard [Win32 Error Codes](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d). These are neither hierarchical nor have any special ordering. Basically it is the same as elektra has now except for no duplicated
+    errors.
+* Mac OS X plist:
+    Just like Windows, plist uses standard Mac OS X errors which is a [huge catalog](http://krypted.com/lists/comprehensive-list-of-mac-os-x-error-codes/) of unordered
+    return codes as integers.
+* [SNMP Standard](http://www.snmp.com/protocol/):
+    Being a standard network protocol, error codes are very specific to the domain itself. A list can be found [here](https://docs.microsoft.com/en-us/windows/desktop/snmp/snmp-error-codes) and would not meet the needs of elektra at all.
+* POSIX:
+    Returning a non-zero value and retrieving the concrete information from `errno` would not suffice for elektra as it is too simple. It would not solve any of our current
+    problems like having excessive uncategorized codes for errors.
+
+## Decision
+
+The error message has the current format:
+
+```
+The command kdb set failed while accessing the key database with the info:
+Sorry, the error (#121) occurred ;(
+Description: validation failed
+Reason: Validation of key "<key>" with string "<value>" failed.
+Ingroup: plugin
+Module: enum
+At: ....../src/plugins/enum/enum.c:218
+Mountpoint: <parentKey>
+Configfile: ...../<file>.25676:1549919217.284067.tmp
+```
+
+The new default error message will look like this:
+
+```
+Sorry, plugin <PLUGIN> issued error #<NR>:
```

Comment:
Since the new error is a String I would not prefix it with a letter.
Why not take a more straightforward approach and use `... issued error code XXXXX`?

Reply:
Because people copy&paste only the XXXXX and sometimes the information if it was a warning/error gets lost then. Please also create an issue about that.

Reply:
Well, the concept relies on the fact that errors and warning are interchangeable like you requested. It also does not make a considerable difference if it is a Resource "Warning" or "Error" for example. The message itself states if it was an error or warning and is even color highlighted. Honestly I would not prefix a code with "W" or "E" just because user could potentially lazy copy an error code. The error code itself also is primarily used for programmatic reaction rather than for users to copy. Users should rely on the message. This is what we have talked all about in this PR

Reply:
Ok.